### PR TITLE
Expose icon mouseover state to parent component

### PR DESCRIPTION
**Pull request by chukwuemeka**
Proposed enhancement to cater for issue #344

### CODE REVIEW

File: src/BurgerIcon.js [hunk 1]
```diff
@@ -79,8 +79,16 @@ export default class BurgerIcon extends Component {
         {icon}
         <button
           onClick={this.props.onClick}
-          onMouseOver={() => this.setState({ hover: true })}
-          onMouseOut={() => this.setState({ hover: false })}
+          onMouseOver={() => {
+            this.setState({ hover: true });
+            if (this.props.onIconStateChange)
+              this.props.onIconStateChange({ isMouseIn: true });
```
Comment (negomi) [inline src/BurgerIcon.js]: My only suggestion would be changing the name to `onIconHoverChange` to make it more obvious it's just triggered on the hover event.

Comment (chukwuemeka) [reply src/BurgerIcon.js]: Thanks! Will do that now.

Comment (chukwuemeka) [reply src/BurgerIcon.js]: Done.

Comment (negomi) [reply src/BurgerIcon.js]: Thanks!

**Pull request merged**